src/napari_spatialdata/_widgets.py

```diff
@@ -232,6 +240,65 @@ def _(self, vec: pd.Series, key: str, layer: Layer) -> dict[str, Any]:
         }
 
 
+class ScatterListWidget(AListWidget):
+    attrChanged = Signal()
+    _text = None
+
+    def __init__(self, viewer: Viewer, model: ImageModel, attr: str, **kwargs: Any):
+        AListWidget.__init__(self, viewer, model, attr, **kwargs)
+        self.attrChanged.connect(self._onChange)
+        self.data = -1
+
+    def _onAction(self, items: Iterable[str]) -> None:
+        for item in sorted(set(items)):
+            try:
+                vec, _ = self._getter(item, index=self.getIndex())
+            except Exception as e:  # noqa: B902
+                logger.error(e)
+                continue
+            self.data = vec
+        print("Selected data: ", self.data)
+        return
+
+    def getData(self) -> NDArrayA:
+        # check type?
+        return self.data
+
+    def setAttribute(self, field: Optional[str]) -> None:
+        if field == self.getAttribute():
+            return
+        if field not in ("var", "obs", "obsm"):
+            raise ValueError(f"{field} is not a valid adata field.")
+        self._attr = field
+        self._getter = getattr(self.model, f"get_{field}")
+        self.attrChanged.emit()
+
```

Comment:
can we use setters and getters also here?

Reply:
I mean both for `setAttribute` and `getAttribute`

Reply:
ok. I wrote these based on setindex and getindex functions in the alistwdget class for self._index, should it also be changed there for property setter and getter or is it a different situation?

Reply:
good point, because of the attribute getter and emit maybe ok to not have as property